### PR TITLE
Add codespell integration for spell checking

### LINKED ISSUE #4555: setup codespell to avoid more typos
There are currently tons of typos in this project. Someone should fix them.
Then setup codespell spell to check for them each time there is a pr 👍 

https://github.com/codespell-project/codespell

image

### PR DESCRIPTION
**Pull request by aroramrinaal**
This pull request adds codespell integration to the project, enabling automated spell checking for common misspellings in the source code. By running codespell on each pull request targeting the gh-pages branch.

closes #4555

### CODE REVIEW

**Comment by shiftkey** [comment]
Can we go even simpler and just use the [Action for codespell](https://github.com/codespell-project/actions-codespell)?

File: .github/workflows/codespell.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  pull_request:
+    branches:
+      - gh-pages
+
+jobs:
+  codespell:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (shiftkey) [inline .github/workflows/codespell.yml]: We should use a newer version of the checkout command:

```suggestion
        uses: actions/checkout@v4
```

**Comment by shiftkey** [comment]
Looks like at least `assets/upforgrabs.ai` should be ignored by codespell, as that's not a text file

**Comment by RayBB** [comment]
@aroramrinaal can you update the PR with these suggestions?

**Comment by shiftkey** [comment]
Closing in favour of https://github.com/up-for-grabs/up-for-grabs.net/pull/4703 which gets this working on CI and addresses regressions

**Pull request closed without merging**